parquet-column/src/test/java/org/apache/parquet/io/api/TestBinary.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.io.api;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.nio.ByteBuffer;
+
+public class TestBinary {
+  @Test
+  public void testSliceForByteArrayBackedBinary() {
+    Binary binary = Binary.fromConstantByteArray("test-123".getBytes());
+    Assert.assertArrayEquals("123".getBytes(), binary.slice(5, 3).getBytesUnsafe());
+  }
+
+  @Test
+  public void testSliceForByteArraySliceBackedBinary() {
```

Comment:
how about make some tests that all Binary's must pass, and then run all the Binary instances through them?

eg:

```
interface BinaryFactory { Binary get(String s); }

private void testSlice(BinaryFactory bf) {
  Binary b = bf.get("test-123");
  assertArrayEquals("123".getBytes("UTF-8"), b.slice(5, 3).getBytesUnsafe());
}

private void testCopy(BinaryFactory bf) {
 ...
}

private void testFoo(BinaryFactory bf) {
 ...
}

private void testBinary(BinaryFactory bf) {
  testSlice(bf);
  testCopy(bf);
  testFoo(bf);
}

@Test
pubic void testConstantBinary() {
  testBinary(new BinaryFactory { @Override Binary get(String s) { return Binary.fromConstantByteArray(s.getBytes("UTF-8")) } });
}

@Test
pubic void testBufferBinary() {
  testBinary(new BinaryFactory { @Override Binary get(String s) { return Binary.fromBuffer(...) } });
}

etc.
```